### ISSUE TITLE
Write a "Copyright and Citing" page for Doxygen.

### ISSUE THREAD

**Issue by joergi-w** [seqan/seqan3#781]
It should explain how to cite SeqAn3 and how it can be redistributed. 

Also add rules for the **snippets**. h-2's suggests to use:
```
// This file is licensed under CreativeCommons - Zero 1.0 Universal / released into the Public Domain.
// You may use this file as wish, including removing this copyright notice.
// This does not affect obligations to include a notice to SeqAn3's copyright when distributing the SeqAn3 library.
```
(see discussion in #780)

**Comment by rrahn** [comment]
We would in general need links to the seqan website and the github repo etc. on the documentation.

**Comment by h-2** [comment]
resolved by #928
